src/main/java/org/carlspring/cloud/storage/s3fs/S3FileSystemProvider.java

```diff
@@ -958,6 +958,17 @@ public Properties loadAmazonProperties()
     boolean exists(S3Path path)
     {
         S3Path s3Path = toS3Path(path);
+
+        if(isBucketRoot(s3Path)) {
+            // check to see if bucket exists
+            try {
+                s3Utils.listS3Objects(s3Path);
+                return true;
+            } catch (SdkException e) {
+                return false;
+            }
+        }
```

Comment:
Cosmetic: Just braces on new lines, please.
```suggestion
        if(isBucketRoot(s3Path))
        {
            // check to see if bucket exists
            try
            {
                s3Utils.listS3Objects(s3Path);
                return true;
            }
            catch (SdkException e)
            {
                return false;
            }
        }
```